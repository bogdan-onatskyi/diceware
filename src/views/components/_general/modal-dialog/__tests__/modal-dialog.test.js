import React from 'react';

import ModalDialog from '../modal-dialog';
import Button from '../../button/button';

describe("ModalDialog component", () => {
    it("should match saved snapshot (type === 'sometype')", () => {
        const handleClose = jest.fn();
        const wrapper = shallow(
            <ModalDialog title="Title" isOpen={true} handlerClose={handleClose}>
                <span>text</span>
            </ModalDialog>
        );
        expect(wrapper).toMatchSnapshot();
        wrapper.find('.modal__title--close-button').simulate('click');
        expect(handleClose).toHaveBeenCalled();

        wrapper.find(Button).simulate('click');
        expect(handleClose).toHaveBeenCalled();
    });

    it("should have 3 props", () => {
        const propsStr = 'title isOpen handlerClose';
        const props = [];
        for (let prop in ModalDialog.propTypes) props.push(prop);
        expect(props.join(' ') === propsStr).toBe(true);
    });
});
