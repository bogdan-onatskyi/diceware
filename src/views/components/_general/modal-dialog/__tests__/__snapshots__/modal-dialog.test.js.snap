// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDialog component should match saved snapshot (type === 'sometype') 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={0}
  contentLabel="onRequestClose Example"
  isOpen={true}
  onRequestClose={[Function]}
  overlayClassName="overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={false}
  shouldFocusAfterRender={true}
>
  <div
    className="modal__title"
  >
    <span
      className="modal__title--content"
    >
      Title
    </span>
    <span
      className="modal__title--close-button"
      data-tip="Закрыть"
      onClick={[Function]}
    >
       × 
    </span>
  </div>
  <div
    className="modal__content"
  >
    <div
      className="modal__content--children"
    >
      <span>
        text
      </span>
      <Button
        onClick={[Function]}
        type="dialog-close"
      >
        Закрыть
      </Button>
    </div>
  </div>
  <ToolTip />
</Modal>
`;
